packages/ui/react-ui-stack/src/components/Section.tsx

```diff
@@ -76,19 +79,48 @@ export const Section = forwardRef<HTMLLIElement, SectionProps>(
             <div role='none' className='flex flex-1 min-is-0 overflow-hidden'>
               {children}
             </div>
-
-            <Button
-              variant='ghost'
-              classNames={[
-                'self-stretch justify-start rounded-is-none',
-                hoverableFocusedControls,
-                active === 'destination' && 'invisible',
-              ]}
-              onClick={onRemove}
+            <DropdownMenu.Root
+              {...{
+                open: optionsMenuOpen,
+                onOpenChange: (nextOpen: boolean) => {
+                  // if (!nextOpen) {
+                  //   suppressNextTooltip.current = true;
+                  // }
+                  return setOptionsMenuOpen(nextOpen);
+                },
+              }}
             >
-              <span className='sr-only'>{t('remove section label')}</span>
-              <X className={mx(getSize(4), hoverableControlItem, 'transition-opacity')} />
-            </Button>
+              <DropdownMenu.Trigger asChild>
+                <Button
+                  variant='ghost'
+                  classNames={[
+                    'shrink-0 pli-2 pointer-fine:pli-1 rounded-none',
+                    hoverableControlItem,
+                    hoverableFocusedControls,
+                    hoverableOpenControlItem,
+                    active === 'overlay' && 'invisible',
+                  ]}
+                  // data-testid={testId}
+                >
+                  <DotsThreeVertical className={getSize(4)} />
+                </Button>
+              </DropdownMenu.Trigger>
+
+              <DropdownMenu.Portal>
+                <DropdownMenu.Content classNames='z-[31]'>
```

Comment:
You shouldn't need to set z-index if your content is within a `Portal` (I realize even my own code does this) — this isn't blocking, but could you try removing the z-index and see if it still works as expected?

Reply:
sure! will give it a try.

Reply:
works great